pkg/codegen/hcl2/program.go

```diff
@@ -96,3 +98,18 @@ func (p *Program) NewDiagnosticWriter(w io.Writer, width uint, color bool) hcl.D
 func (p *Program) BindExpression(node hclsyntax.Node) (model.Expression, hcl.Diagnostics) {
 	return p.binder.bindExpression(node)
 }
+
+// Packages returns the list of package schemas used by this program.
+func (p *Program) Packages() []*schema.Package {
+	names := make([]string, 0, len(p.binder.packageSchemas))
+	for name := range p.binder.packageSchemas {
+		names = append(names, name)
+	}
+	sort.Strings(names)
+
+	packages := make([]*schema.Package, len(names))
+	for i, name := range names {
+		packages[i] = p.binder.packageSchemas[name].schema
+	}
```

Comment:
Any reason not to match the format of the previous loop?
```suggestion
	packages := make([]*schema.Package, 0, len(names))
	for _, name := range names {
		packages = append(packages, p.binder.packageSchemas[name].schema)
	}
```

Reply:
Yes: I prefer to use explicit indices when they're trivially available. The previous loop can't use them because the expression being ranged over is a map.